core/src/main/java/io/substrait/relation/Set.java

```diff
@@ -0,0 +1,54 @@
+package io.substrait.relation;
+
+import io.substrait.proto.SetRel;
+import io.substrait.type.Type;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class Set extends AbstractRel {
+    public abstract Set.SetOp getSetOp();
+
+    public static enum SetOp {
+        UNKNOWN(SetRel.SetOp.SET_OP_UNSPECIFIED),
+        MINUS_PRIMARY(SetRel.SetOp.SET_OP_MINUS_PRIMARY),
+        MINUS_MULTISET(SetRel.SetOp.SET_OP_MINUS_MULTISET),
+        INTERSECTION_PRIMARY(SetRel.SetOp.SET_OP_INTERSECTION_PRIMARY),
+        INTERSECTION_MULTISET(SetRel.SetOp.SET_OP_INTERSECTION_MULTISET),
+        UNION_DISTINCT(SetRel.SetOp.SET_OP_UNION_DISTINCT),
+        UNION_ALL(SetRel.SetOp.SET_OP_UNION_ALL);
+
+        private SetRel.SetOp proto;
+
+        SetOp(SetRel.SetOp proto) {
+            this.proto = proto;
+        }
+
+        public SetRel.SetOp toProto() {
+            return proto;
+        }
+
+        public static Set.SetOp fromProto(SetRel.SetOp proto) {
+            for (var v : values()) {
+                if (v.proto == proto) {
+                    return v;
+                }
+            }
+
+            throw new IllegalArgumentException("Unknown operation: " + proto);
+        }
+    }
+
+    @Override
+    protected Type.Struct deriveRecordType() {
+        return getInputs().get(0).getRecordType();
```

Comment:
Is it safe to assume that we have at least one input?
If not, how should I handle a case when all relations are empty?